app/src/main/kotlin/at/bitfire/davdroid/ui/account/AccountScreenModel.kt

```diff
@@ -183,13 +185,26 @@ class AccountScreenModel @AssistedInject constructor(
     }
 
     fun setCollectionSync(id: Long, sync: Boolean) {
-        viewModelScope.launch {
+        viewModelScope.launch(Dispatchers.IO) {
             collectionRepository.setSync(id, sync)
+            syncAfterDelay(id)
         }
     }
 
     fun sync() {
         syncWorkerManager.enqueueOneTimeAllAuthorities(account, manual = true)
     }
 
+    /**
+     * Enqueues a one-time account wide sync after a short delay or scope cancellation.
+     * @param collectionId collection ID of collection in the account to be synchronized
+     */
+    private fun syncAfterDelay(collectionId: Long) {
+        val serviceId = collectionRepository.get(collectionId)?.serviceId ?: return
+        val accountName = serviceRepository.get(serviceId)?.accountName ?: return
+        val account = Account(accountName, context.getString(R.string.account_type))
```

Comment:
Better create the account using `accountRepository.fromName()`, so we don't need to access the string and also have a better start when moving the accounts to the DB.

Reply:
I forgot we had that available.